search-service/src/main/java/com/movie/recommender/search/model/SearchFilter.java

```diff
@@ -0,0 +1,102 @@
+package com.movie.recommender.search.model;
+
+import lombok.Data;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+@Data
+public class SearchFilter implements RatingFilter, DateFilter, TitleFilter, GenresFilter, CastNamesFilter, CrewNamesFilter {
```

Comment:
using interfaces here is overkill
adding interfaces adds unnecessary complexity without clear benefits